### PR TITLE
impl From<Finished> for GameState { ... }

### LINKED ISSUE #463: impl From<Finished> for GameState { ... }
We need to handle the transition from finished if the game controller controller accidentally presses finished and undoes it.

### PR DESCRIPTION
**Pull request by Vivituhh**
## Introduced Changes

We add a new intermediate state `State::TentativeFinished` where the robot is in `GameState::Set` and the GameController controller has 4s to undo a wrongly clicked `Finished`.

Fixes #463 

## Ideas for Next Iterations (Not This PR)
None

## How to Test

Use GameController to test and click `Finished` when it's not normally done.

### CODE REVIEW

File: crates/control/src/game_state_filter.rs [hunk 1]
```diff
@@ -116,23 +116,46 @@ fn next_filtered_state(
...
+            State::from_game_state(game_state)
+        }
+        (
+            State::TentativeFinished {
+                time_when_finished_clicked,
+            },
+            _,
+        ) => State::TentativeFinished {
+            time_when_finished_clicked,
+        },
```
Comment (schluis) [inline crates/control/src/game_state_filter.rs:144]: Can you move this up as `else` case for `if` in line 126?

File: crates/control/src/game_state_filter.rs [hunk 2]
```diff
@@ -324,6 +350,7 @@ impl State {
                 GamePhase::PenaltyShootout { .. } => FilteredGameState::Set,
                 _ => FilteredGameState::Finished,
             },
+            State::TentativeFinished { .. } => FilteredGameState::Set,
```
Comment (schluis) [inline crates/control/src/game_state_filter.rs:353]: Please mark this as hack

**Review by schluis — CHANGES_REQUESTED** [review]
Very nice 🚀

File: crates/control/src/game_state_filter.rs (hunk 1 above)
Comment (Vivituhh) [reply crates/control/src/game_state_filter.rs]: Done

Comment (schluis) [reply crates/control/src/game_state_filter.rs]: Nice 👍🏻

**Review by schluis — APPROVED** [review]

**Pull request merged**